pkg/i2gw/providers/istio/resource_reader.go

```diff
@@ -58,6 +60,25 @@ func (r *reader) readResourcesFromCluster(ctx context.Context) (*storage, error)
 	return &res, nil
 }
 
+func (r *reader) readResourcesFromFile(_ context.Context, filename string) (*storage, error) {
+	stream, err := os.ReadFile(filename)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read file %v: %w", filename, err)
+	}
+
+	unstructuredObjects, err := i2gw.ExtractObjectsFromReader(bytes.NewReader(stream), r.conf.Namespace)
+	if err != nil {
+		return nil, fmt.Errorf("failed to extract objects: %w", err)
+	}
```

Comment:
As a followup to my last comment, I see we're using `ExtractObjectsFromReader` here. Maybe move it to the `common` package?

Reply:
can move it to common yeah, good idea